core/src/main/java/org/arquillian/smart/testing/Logger.java

```diff
@@ -1,67 +1,25 @@
 package org.arquillian.smart.testing;
 
-import java.util.logging.ConsoleHandler;
-import java.util.logging.Level;
-
 import static java.lang.String.format;
 import static org.arquillian.smart.testing.Configuration.SMART_TESTING_DEBUG;
 
 public class Logger {
 
-    private final java.util.logging.Logger jul;
+    private static Logger logger = null;
 
     private static final String PREFIX = "%s: Smart-Testing - ";
 
     private boolean mavenDebugLogLevel = false;
 
-    private Logger(java.util.logging.Logger logger) {
-        this.jul = logger;
-        setLogLevel();
-    }
-
-    /**
-     * Find or create a logger for a named subsystem.
-     *
-     * @param cls
-     *     A class name of the subsystem for the logger such as java.net.URI or javax.swing.Box
-     *
-     * @return a suitable Logger
-     */
-    public static Logger getLogger(Class cls) {
-        return getLogger(cls.getName());
-    }
+    private Logger() {
 
-    /**
-     * Delegates creation of a logger for a named subsystem to java.util.logging.Logger.
-     *
-     * @param name
-     *     A name for the logger.  This should be a dot-separated name and should normally
-     *     be based on the package name or class name of the subsystem, such as java.net
-     *     or javax.swing
-     *
-     * @return a suitable Logger
-     *
-     * @throws NullPointerException
-     *     if the name is null.
-     */
-    public static Logger getLogger(String name) {
-        return new Logger(java.util.logging.Logger.getLogger(name));
     }
 
-    /**
-     * Log a formatted message with given arguments using java.util.logging.Logger
-     *
-     * @param level
-     *     One of the message level identifiers, e.g., SEVERE
-     * @param msg
-     *     The string message (or a key in the message catalog)
-     * @param args
-     *     arguments to the message
-     */
-    public void log(Level level, String msg, Object... args) {
-        if (jul.isLoggable(level)) {
-            jul.log(level, getFormattedMsg(msg, args));
+    public static Logger getLogger() {
+        if (logger == null) {
+            logger = new Logger();
```

Comment:
Let's just return new instance for each and every call to this factory method. We don't have to share the instance across the classes.